client/mysqldump.c

```diff
@@ -3994,7 +3998,7 @@ static int dump_triggers_for_table(char *table_name, char *db_name)
 done:
   if (path)
     my_fclose(sql_file, MYF(0));
-
+  mysql_free_result(show_triggers_rs);
```

Comment:
This is another case that could cause a crash. Consider this code above:

```c
 if (mysql_query_with_error_report(mysql, &show_triggers_rs, query_buff))
    goto done;
```
This will not fill `show_triggers_rs` with anything if the result is not `0`. `show_triggers_rs` is not initialised, so there is a case now where a failure here could jump to `done` and run `mysql_free_result` on uninitialised memory.